src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -255,7 +274,25 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
         // only update date value if not being set by user
         filter(() => !this.datepickerFocusService.elementIsFocused(this.el.nativeElement))
       )
-      .subscribe((value: string) => this.updateDate(this.dateIOService.getDateValueFromDateString(value)));
+      .subscribe((value: string) => {
+        if (value === '' && this.previousDateChange === null) {
+          // when value is empty string and previous state is null
+          // no new date could be produced so there is no need of any
+          // actions.
+          return;
+        }
+        const input = this.dateIOService.getDateValueFromDateString(value);
+        if ((this.previousDateChange === null || this.previousDateChange === undefined) && input === null) {
+          // when the new date will result into null and previous value is null
+          // no need to update
+          return;
+        }
+
+        // toString() let me compare Date .. milliseconds are a thing!
+        if ((input && input.toString()) !== (this.previousDateChange && this.previousDateChange.toString())) {
```

Comment:
Might be better to use the new `toComparisonString()` method so you get a `YYYYMMDD` string without the time values.

Reply:
Sounds, better will do!

Reply:
After testing it seems that the `input` could be:

* Date
* String
* empty string

So it's getting pretty complex to use `new DayModel` and converting `input` into YYYY-MM-DD.

So `toString()` is doing this a lot faster and simple  - so prefer to keep it like it is.

Reply:
I think there might be edge cases, where dates have timezone conflicts. There was another bug with this problem recently, so I would prefer to make `input` be a DayModel and compare the same way. Might be good to hoist the comparison logic to one place since its being duplicated.